### PR TITLE
Feature/multispeaker

### PR DESCRIPTION
**Pull request by cschaefer26**
- implement optional multispeaker model and preprocessing
- remove WaveRNN vocoder

### CODE REVIEW

File: utils/dataset.py [hunk 1]
```diff
@@ -342,47 +275,67 @@ def get_tts_datasets(path: Path,
     return train_set, val_set
 
 
-def get_binned_taco_dataloader(data_path: Path, max_batch_size: int = 8) -> BinnedTacoDataLoader:
-    train_data = unpickle_binary(data_path / 'train_dataset.pkl')
-    val_data = unpickle_binary(data_path / 'val_dataset.pkl')
+def get_binned_taco_dataloader(paths: Paths, max_batch_size: int = 8) -> BinnedTacoDataLoader:
+    train_data = unpickle_binary(paths.data / 'train_dataset.pkl')
```
Comment (alexteua) [inline utils/dataset.py:279]: These values for filenames are hardcoded in multiple places. How about to create entries for them as well in the Paths ?

File: train_tacotron.py [hunk 2]
```diff
@@ -167,25 +179,19 @@ def create_align_features(model: Tacotron,
     train_cfg = config['tacotron']['training']
     if args.force_gta:
```
Comment (alexteua) [inline train_tacotron.py:180]: There is no check for passed arguments, for example one can pass extract_pitch and force_gta and after extracting pitch the process will finish. It creates some confusion to have training as one of the options in the if-else block with other operations. I would suggest to keep only training logic in this script, and move force_align force_gta extract_pitch to another helper script with own arguments and execution flow

File: train_forward.py [hunk 3]
```diff
@@ -81,11 +81,14 @@ def create_gta_features(model: Tacotron,
 
     if force_gta:
         print('Creating Ground Truth Aligned Dataset...\n')
```
Comment (alexteua) [inline train_forward.py:83]: Similar to the comment in the train_tacotron.py, I would suggest to move force_gta functionality out of this sctipt or at least not to mix within the same if-else conctruct with training options

File: preprocess.py [hunk 4]
```diff
@@ -1,16 +1,32 @@
...
+from pitch_extraction.pitch_extractor import PitchExtractor, new_pitch_extractor_from_config
 from utils.display import *
 from utils.dsp import *
 from utils.files import get_files, pickle_binary, read_config
 from utils.paths import Paths
 from utils.text.cleaners import Cleaner
-from utils.text.recipes import ljspeech
+
+
+SPEAKER_EMB_DIM = 256
```
Comment (alexteua) [inline preprocess.py:29]: Can we read this from the config?

File: preprocess.py [hunk 5]
```diff
@@ -36,123 +52,178 @@ def __init__(self,
...
 
-    for i, dp in enumerate(pool.imap_unordered(preprocessor, wav_files), 1):
-        if dp is not None and dp.item_id in text_dict:
-            dataset += [(dp.item_id, dp.mel_len)]
-            cleaned_texts += [(dp.item_id, dp.text)]
-        bar = progbar(i, len(wav_files))
-        message = f'{bar} {i}/{len(wav_files)} '
-        stream(message)
+    device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+    voice_encoder = VoiceEncoder().to(device)
```
Comment (alexteua) [inline preprocess.py:173]: nice that now preprocessing will utilize GPU!

**Review by alexteua — CHANGES_REQUESTED** [review]
Some comments after the first pass through changes :)

File: utils/dataset.py (hunk 1 above)
Comment (cschaefer26) [reply utils/dataset.py]: Good idea, moved to paths.

File: train_tacotron.py (hunk 2 above)
Comment (cschaefer26) [reply train_tacotron.py:180]: I agree but I would like to do that in a separate refactoring - otherwise the changes are getting even larger.

File: train_forward.py (hunk 3 above)
Comment (cschaefer26) [reply train_forward.py:83]: Same as above, lets do that in a separate refactoring.

File: preprocess.py (hunk 4 above)
Comment (cschaefer26) [reply preprocess.py:29]: Could be done but the dim is fixed due to the external dependency of resemblyzer. Added a comment.

**Review by alexteua — APPROVED** [review]
